doc/mgmt/SONiC_TIMEZONE_HLD.md

```diff
@@ -0,0 +1,546 @@
+
+   ```
+     
+#### 1.1.2.2 set time and date 
+Details to be added. 
+    
+### 1.1.3 Functional Requirements
+
+Provide management framework support to          
+- configure timezone     
+- query timezone     
+- configure time and date    
+- query time and date     
+    
+### 1.1.4 Configuration and Management Requirements
+- CLI style configuration and show commands    
+- REST API support    
+- gNMI Support    
+           
+### 1.1.5 Configurations not supported by this feature using management framework:
+- support configuration of timezone abbreviations with offset to UTC        
+- support get timezone from dhcp option           
+           
+### 1.1.6 Scalability Requirements
+No specific scalability requirement for this feature.
+
+### 1.1.7 Warm Boot Requirements
+Configured timezone should survive warm boot. No specific handling is required.    
+    
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+Implement timezone configuration using transformer in sonic-mgmt-framework.             
+
+### 1.2.2 Container
+The front end code changes in the management-framework container includes:  
+     
+**Timezone:**    
+    
+- XML file for the CLI    
+- Python script to handle CLI request (actioner)    
+- Jinja template to render CLI output (renderer)    
+- front-end code to support "show running-configuration"    
+- OpenConfig YANG model in openconfig-system.yang to set and get configured timezone in the configDB    
+- OpenConfig RPC YANG model to get timezone from timedatectl output, including UTC offset    
+- SONiC clock model in sonic-system-clock.yang to set and get configured timezone in the configDB       
+- SONiC clock model in sonic-system-clock.yang to get timezone from timedatectl output, including UTC offset     
+       
+**time and date**   
+     
+Details to be added.      
+      
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases
+Manage/configure timezone via gNMI, REST and CLI interfaces.    
+Manage/configure time&date via gNMI, REST and CLI interfaces.    
+           
+## 2.2 Functional Description
+Provide CLI, gNMI and REST supports for NTP related configurations.
+            
+## 2.3 Backend change to support new configurations
+**timezone**       
+     
+- transformer functions to    
+   * set "system" as key for CLOCK table in CONFIG DB   
+   * rpc function to get output of timedatectl for OpenConfig YANG    
+   * rpc function to get output of timedatectl for SONiC YANG      
+   * subscribe to listen for CLOCK change in CONFIG DB    
+   * set local timezone for glog   
+- customer cvl validation function to reject timezone if the timezone name does not exist under /usr/share/zoneinfo. This is for the case when timezone configuration is done from SONiC YANG.     
+- glog patch to handle local timezone change for the timestamp    
+- SONiC click CLI enhancement if possible.         
+    
+**time and date**   
+     
+Details to be added.     

```

Comment:
Any TODO here?

Reply:
For "time and date" set,  it will be added once timezone is completed.